lib/msal-core/src/ServerRequestParameters.ts

```diff
@@ -294,11 +245,16 @@ export class ServerRequestParameters {
      * Utility to generate a QueryParameterString from a Key-Value mapping of extraQueryParameters passed
      * @param extraQueryParameters
      */
-    static generateQueryParametersString(queryParameters: StringDict): string {
+    static generateQueryParametersString(queryParameters: StringDict, silentCall?: boolean): string {
         let paramsString: string = null;
 
         if (queryParameters) {
             Object.keys(queryParameters).forEach((key: string) => {
+                // sid cannot be passed along with login_hint or domain_hint
+                if(key == "domain_hint" && (silentCall || paramsString.match("sid"))) {
```

Comment:
Is this going to work? paramsString is null the first time through the loop. Maybe check if the "sid" key exists in queryParameters instead?

Reply:
Thanks for the alert, this is a typo while checking in, will do the needful changes.

Reply:
@sameerag I have tried this PR with my B2C custom policy and have a need to pass both domain_hint and login_hint. Would you explain why you are not at least allowing the developer to pass domain_hint in extraQueryParameters if they so desired. I have policy logic that is using the domain_hint. As discussed elsewhere I do also need to pass login_hint

Reply:
@alex-mason-jdas I have forwarded this requirement to B2C service team and will get back once I hear from them. As stated in the PR description, we made these changes as per AAD service recommendation and may be B2C has a different set of requirements. Will reach out once we have an answer.